### PR TITLE
fix(grpc_client): pass ResponsesRequest sampling params to all backends

### PR DESCRIPTION
**Pull request by key4ng**
### Problem

ResponsesRequest sampling parameters (`top_k`, `min_p`, `frequency_penalty`, `presence_penalty`, `repetition_penalty`) were hardcoded to default values in all three gRPC backend builders (vLLM, SGLang, TRT-LLM). User-supplied values were silently ignored.

### Solution

Wire the sampling parameters from `ResponsesRequest` through to each backend's proto builder:
- **vLLM**: negative `top_k` clamped to 0 (vLLM's disabled sentinel), cast to `u32`
- **SGLang**: `top_k` passed as-is (`-1` = disabled)
- **TRT-LLM**: negative `top_k` clamped to 0, `min_p` and penalties wired through
- **Stop sequences**: annotated as not yet passed through (follow-up fix)

## Changes

- `crates/grpc_client/src/vllm_engine.rs` — pass sampling params from `ResponsesRequest` instead of hardcoded defaults
- `crates/grpc_client/src/sglang_scheduler.rs` — same
- `crates/grpc_client/src/trtllm_service.rs` — same
- `e2e_test/responses/test_sampling_params.py` — E2E tests for sampling parameter pass-through (non-streaming + streaming + deterministic temperature=0)
- Unit tests added to each backend for round-trip assertion and disabled-sentinel edge cases

## Test Plan

- [x] Unit tests for all three backends verify field values and edge cases (negative top_k clamping)
- [x] E2E test suite covers non-streaming, streaming, and temperature=0 determinism
- [x] `cargo test -p smg-grpc-client` passes

<details>
<summary>Checklist</summary>

- [x] `cargo +nightly fmt` passes
- [x] `cargo clippy --all-targets --all-features -- -D warnings` passes

</details>

## Summary by CodeRabbit

* **New Features**
  * Request sampling parameters (top_k, min_p, frequency_penalty, presence_penalty, repetition_penalty) are now honored end-to-end across gRPC backends; sentinel/default handling preserved where applicable.

* **Tests**
  * Added comprehensive end-to-end tests verifying sampling-parameter propagation, streaming behavior, and determinism for temperature=0.
  * Added unit tests validating sampling-parameter pass-through and default/disabled sentinel behavior.

### CODE REVIEW

File: crates/grpc_client/src/trtllm_service.rs [hunk 1]
```diff
@@ -594,7 +594,7 @@ impl TrtllmServiceClient {
         proto::SamplingConfig {
             beam_width: 1,
             num_return_sequences: 1,
-            top_k: None,
+            top_k: Some(request.top_k.max(0)),
```
[the remark this answers is not in the document]
Comment (key4ng) [reply crates/grpc_client/src/trtllm_service.rs]: Good catch — fixed in c232a376. Now uses `(request.top_k >= 0).then_some(request.top_k)` so default `-1` maps to `None`, matching the chat builder's `request.top_k.map(|v| v.max(0))` pattern.

File: crates/grpc_client/src/trtllm_service.rs [hunk 2]
```diff
@@ -603,14 +603,14 @@ impl TrtllmServiceClient {
...
-            frequency_penalty: None,
+            repetition_penalty: Some(request.repetition_penalty),
+            presence_penalty: request.presence_penalty,
+            frequency_penalty: request.frequency_penalty,
             prompt_ignore_length: None,
             length_penalty: None,
             early_stopping: None,
             no_repeat_ngram_size: None,
-            min_p: None,
+            min_p: Some(request.min_p),
```
[the remark this answers is not in the document]
Comment (key4ng) [reply crates/grpc_client/src/trtllm_service.rs]: Fixed in the same commit — uses `(request.min_p != 0.0).then_some(request.min_p)` so default `0.0` maps to `None`, matching the chat builder's `min_p: request.min_p` (which is already `Option<f32>`).

File: crates/grpc_client/src/trtllm_service.rs [hunk 3]
```diff
@@ -1009,6 +1009,39 @@ mod tests {
...
+        assert_eq!(cfg.frequency_penalty, Some(0.3));
+        assert_eq!(cfg.presence_penalty, Some(-0.4));
+
+        // Negative top_k is clamped to 0 (TRT-LLM's disabled sentinel).
+        let disabled = ResponsesRequest {
+            top_k: -1,
+            ..Default::default()
+        };
+        let disabled_cfg = TrtllmServiceClient::build_sampling_config_from_responses(&disabled);
+        assert_eq!(disabled_cfg.top_k, Some(0));
```
[the remark this answers is not in the document]
Comment (key4ng) [reply crates/grpc_client/src/trtllm_service.rs]: Updated — test now asserts `disabled_cfg.top_k == None` and `disabled_cfg.min_p == None`.

File: e2e_test/responses/test_sampling_params.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            top_p=0.9,
+            max_output_tokens=256,
+            extra_body={
+                "top_k": 40,
+                "min_p": 0.05,
+                "frequency_penalty": 0.5,
+                "presence_penalty": 0.3,
+                "repetition_penalty": 1.1,
+            },
+        )
```
[the remark this answers is not in the document]
Comment (key4ng) [reply e2e_test/responses/test_sampling_params.py:35]: Skipping — two occurrences in a small test file. The duplication is clear and the extra abstraction isn't worth the indirection for 3 tests.

File: e2e_test/responses/test_sampling_params.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        assert resp.status == "completed"
+        assert resp.error is None
+        assert len(resp.output) > 0
+
+        assert resp.temperature == pytest.approx(0.5)
+        assert resp.top_p == pytest.approx(0.9, abs=1e-4)
+        assert resp.max_output_tokens == 256
+
+        assert resp.usage is not None
+        tokens = resp.usage.output_tokens or resp.usage.completion_tokens
```
[the remark this answers is not in the document]
Comment (key4ng) [reply e2e_test/responses/test_sampling_params.py:45]: The SDK's `ResponseUsage` does include `completion_tokens` — confirmed by the actual test output: `ResponseUsage(input_tokens=None, ..., completion_tokens=8)`. The `or` fallback works as intended. The `output_tokens == 0` edge case doesn't apply here since we're guarding against `None`, not zero.

File: e2e_test/responses/test_sampling_params.py [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+        completed = [e for e in events if e.type == "response.completed"]
+        assert len(completed) == 1
+
+        resp = completed[0].response
+        assert resp.status == "completed"
+        assert len(resp.output) > 0
+        assert resp.temperature == pytest.approx(0.5)
+        assert resp.top_p == pytest.approx(0.9, abs=1e-4)
+        assert resp.usage is not None
+        tokens = resp.usage.output_tokens or resp.usage.completion_tokens
```
[the remark this answers is not in the document]
Comment (key4ng) [reply e2e_test/responses/test_sampling_params.py:91]: Same as above — `completion_tokens` exists on the response object. No change needed.

**Review by CatherineSue — APPROVED** [review]
